src/project.c

```diff
@@ -1092,11 +1093,17 @@ char *getTmpName(char *fname)
     // --- for non-Windows systems:
 #else
     // --- use system function mkstemp() to create a temporary file name
+/*    
     int f = -1;
     strcpy(fname, "enXXXXXX");
     f = mkstemp(fname);
     close(f);
     remove(fname);
+*/
+    strcpy(fname, "enXXXXXX");
+    FILE *f = fdopen(mkstemp(fname), "r");
```

Comment:
we should consider checking for `mkstemp` returning a `-1` error - and subsequently a `NULL` from `fdopen`

Reply:
The latest commit checks if `fdopen `returns `NULL `(which will happen if `mkstemp `returns -1 indicating failure). In this case it returns an empty string (not `NULL`!) for the temporary file name. The same is now done for Windows in case `_tempnam` fails. Any attempt to subsequently open the empty-named file will trigger an EPANET series 300 error.

The fix to `xstrcpy `was in response to a DLL crash when testing one of my older networks with demand categories names longer than the allowed 31 characters. However it appears that it is now causing the test cases that use the API to replace comment strings to fail.